rock/runtime.rb

```diff
@@ -252,11 +254,30 @@ def start_broadcaster(name = Transformer.broadcaster_name, options = Hash.new)
             else
             	Transformer.warn "Transformer broadcaster was already running. Reusing existing task context"
             end
-            
+
+            if block_given?
             Orocos::Process.run(options) do
                 @broadcaster = Orocos.name_service.get(name)
                 yield
             end
+            else
+                start_time = Time.new
+                Thread.new do
+                    Orocos::Process.run(options) do
```

Comment:
Orocos.run can be called without a block. You want to simply do

~~~
@broadcaster = Orocos.run(options)
~~~